feat(v2): add stronger validation for esrvcId and Twilio credential inputs

karrui opened the pull request:
## Problem
<!-- What problem are you trying to solve? What issue does this close? -->
Cleaned up esrvcId and Twilio credential inputs. Trim and add validation for them.

Related to #4372, not sure if I missed any tbh.

## Solution
<!-- How did you solve the problem? -->
**Features**:

- feat: use rhf to add proper validation to esrvcId input
- feat: correctly trim/clear/update twilio credentials on update/delete
- feat: add toggle to show/hide API secret during input

karrui committed:
feat: use rhf to add proper validation to esrvcId input

karrui committed:
feat: correctly trim/clear/update twilio credentials on update/delete

karrui committed:
feat: add toggle to show/hide API secret during input

timotheeg commented on frontend/src/features/admin-form/settings/components/TwilioSettingsSection/TwilioDetailsInputs.tsx:
```diff
@@ -26,48 +38,78 @@ const TWILIO_INPUT_RULES: Record<keyof TwilioCredentials, RegisterOptions> = {
       message: 'Messaging service SID must start with MG',
     },
+    validate: {
+      noWhitespace: (value) =>
+        !value.trim().match(/\s/) ||
+        'Messaging service SID must not contain whitespace',
```
/nit: The `trim()` implies we want to allow accidental leading or trailing whitespaces, but the pattern check above disallows leading whitespace 🤔 . We could probably remove the pattern check to do it in the validator instead, or add optional leading whitespaces in the pattern (e.g. `/^\s*MG/`). And same comment for the checks above.

Also nit: all 4 validators are nearly identical pure functions (minus the error message), so we could have a generator in the module for them.

karrui replied:
inlined whitespace check into pattern validator: 353ba39

timotheeg replied:
Thanks!

Although, the regex updates shouldn't check for trailing whitespace, since there is no tail anchor.

karrui committed:
ref: inline whitespace check into pattern validator

timotheeg reviewed (commented):
Oh, I didn't realize you had dropped the validator to shift the whole check to a pattern. The patterns are not equivalent to the previous code in their current state, since they'd allow a string like this: `  MG. 1. A.  `, which even after `.trim()` wwould contain whitespaces (which the validator was previously disallowing).

We can make this work with patten alone, but I'd recommend using this then:
```javascript
/^\s*MG\S+\s*$/
```
(and similar change for the other regexes)

This guarantees:
1. only leading and trailing whitespaces allowed (they'll be trimmed later when grabbing the value)
2. at least one character of "value" beyond the mandatory prefix

If we know more about the IDs (numeric only? alphanum only? fixed number of characters?), we could replace `\S+` with something more specific too.

karrui committed:
fix: use more specific whitespace regex

allows strings than can be trim()'d to pass but not whitespace in between the input

karrui commented:
> If we know more about the IDs (numeric only? alphanum only? fixed number of characters?), we could replace `\S+` with something more specific too.

unfortunately twilio does not give any guarantees on the shape of those inputs. 
add new regex in 83e4d44

karrui committed:
feat: only show show/hide secret button when adding creds

timotheeg reviewed (approved):
lgtm
